stan/math/prim/scal/fun/init_threadpool_tbb.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef STAN_MATH_PRIM_SCAL_INIT_THREADPOOL_TBB_HPP
+#define STAN_MATH_PRIM_SCAL_INIT_THREADPOOL_TBB_HPP
+
+// TODO(SW): remove STAN_THREADS guards once Intel TBB is fully
+// mandatory
+
+#include <stan/math/prim/scal/fun/get_num_threads.hpp>
+
+#ifdef STAN_THREADS
+#include <tbb/task_scheduler_init.h>
+#endif
+
+namespace stan {
+namespace math {
+
+/**
+ * Initialize the Intel TBB threadpool through the
+ * tbb::task_scheduler_init object. In case an instance of the
+ * tbb::task_scheduler_object is instantiated, then this has no effect,
+ * which is the default behavior of the Intel TBB library.
+ *
+ * The maximal number of threads is read from the environment variable
+ * STAN_NUM_THREADS using internal::get_num_threads. See conventions
+ * of get_num_threads.
+ *
+ * The function returns as a boolean if the tbb::task_scheduler_init
+ * has become active once constructed. The instance may not become
+ * active if there is already another instance created before calling
+ * this function.
+ *
+ * @param stack_size sets the stack size of each thread; the default 0
+ * let's the TBB choose the stack size
+ * @return active status of static tbb::task_scheduler_init
+ * @throws std::runtime_error if the value of STAN_NUM_THREADS env. variable
+ * is invalid
+ */
+inline bool init_threadpool_tbb(tbb::stack_size_type stack_size = 0) {
```

Comment:
Just so I'm clear, `stack_size` set the amount of memory each thread has at initialization right? Is there a reason to say 0 as the default and not something like 64 just to give them a bit?

Reply:
0 let's the TBB choose that initial size. We should leave the default here as is, I think.

Reply:
Gah totally missed that in the docs my bad